Plugins/ExternalTexture/Source/ExternalTexture_D3D12.cpp

```diff
@@ -129,74 +131,63 @@ namespace
 
 namespace Babylon::Plugins
 {
-    class ExternalTexture::Impl
+    class ExternalTexture::Impl final : public ImplBase
     {
     public:
-        Impl(Graphics::TextureT ptr)
+        Impl(Graphics::TextureT ptr);
+
+        void Update(Graphics::TextureT ptr);
+
+        uintptr_t Ptr() const
+        {
+            return reinterpret_cast<uintptr_t>(m_ptr.get());
+        }
+
+    private:
+        void Init(Graphics::TextureT ptr)
         {
             m_ptr.copy_from(ptr);
```

Comment:
This order seems weird for the update case, because you will actually have an instance of the `ExternalTexture` and the m_ptr will be updated but it will also throw. It seems like the copy_from should happen after the throw (doesn't happen).

Reply:
I agree, but then the code will be harder to write.